test/testmathlib.cpp

```diff
@@ -312,6 +313,11 @@ class TestMathLib : public TestFixture {
         ASSERT_EQUALS_DOUBLE(0.0   , MathLib::toDoubleNumber("-0.0"));
         ASSERT_EQUALS_DOUBLE(0.0   , MathLib::toDoubleNumber("+0.0"));
 
+        // Ticket #6795
+        std::setlocale(LC_NUMERIC, "de_DE");
```

Comment:
I just need to change the test case so as to force usage of a locale that has comma as decimal sign, and that's it. The code in Mathlib.cpp reinstates the initial locale right after the call to strtod, so your concerns with libraries we use or users of cppcheck as a lib are covered.